http-clients/src/main/java/org/triplea/http/client/HttpClient.java

```diff
@@ -86,12 +85,12 @@ private static Exception errorDecoder(final String methodKey, final Response res
         .map(body -> {
           try (BufferedReader reader = new BufferedReader(body.asReader())) {
             final String errorMessageBody = reader.lines().collect(Collectors.joining("\n"));
-            return new HttpCommunicationException(response.status(), firstLine + "\n" + errorMessageBody);
+            return new HttpInteractionException(response.status(), firstLine + "\n" + errorMessageBody);
```

Comment:
HttpCommunicationException is renamed to HttpInteractionException, we throw the exception in more cases than just communications errors.